cmd/create/accountroles/creators.go

```diff
@@ -139,35 +148,59 @@ func (up *unmanagedPoliciesCreator) createRoles(r *rosa.Runtime, input *accountR
 		}
 
 		assumeRolePolicy := getAssumeRolePolicy(file, input)
-
-		r.Reporter.Debugf("Creating role '%s'", accRoleName)
 		tagsList := up.getRoleTags(file, input)
-		roleARN, err := r.AWSClient.EnsureRole(accRoleName, assumeRolePolicy, input.permissionsBoundary,
-			input.defaultPolicyVersion, tagsList, input.path, false)
+		filename := fmt.Sprintf("sts_%s_permission_policy", file)
+
+		err := createRoleUnmanagedPolicy(r, input, accRoleName, assumeRolePolicy, tagsList, filename)
 		if err != nil {
 			return err
 		}
-		r.Reporter.Infof("Created role '%s' with ARN '%s'", accRoleName, roleARN)
+	}
 
-		filename := fmt.Sprintf("sts_%s_permission_policy", file)
-		policyPermissionDetail := aws.GetPolicyDetails(input.policies, filename)
+	return nil
+}
 
-		policyARN := aws.GetPolicyARN(r.Creator.AccountID, accRoleName, input.path)
+func (up *unmanagedPoliciesCreator) buildCommands(input *accountRolesCreationInput) (string, error) {
+	commands := []string{}
+	for file, role := range aws.AccountRoles {
+		accRoleName := aws.GetRoleName(input.prefix, role.Name)
+		iamTags := up.getRoleTags(file, input)
 
-		r.Reporter.Debugf("Creating permission policy '%s'", policyARN)
-		if args.forcePolicyCreation {
-			policyARN, err = r.AWSClient.ForceEnsurePolicy(policyARN, policyPermissionDetail,
-				input.defaultPolicyVersion, tagsList, input.path)
-		} else {
-			policyARN, err = r.AWSClient.EnsurePolicy(policyARN, policyPermissionDetail,
-				input.defaultPolicyVersion, tagsList, input.path)
-		}
-		if err != nil {
-			return err
+		createRole := buildCreateRoleCommand(accRoleName, file, iamTags, input)
+
+		policyName := aws.GetPolicyName(accRoleName)
+		policyDocument := fmt.Sprintf("file://sts_%s_permission_policy.json", file)
+
+		createPolicy := buildCreatePolicyCommand(policyName, policyDocument, iamTags, input.path)
+
+		policyARN := aws.GetPolicyARN(input.accountID, accRoleName, input.path)
+
+		attachRolePolicy := buildAttachRolePolicyCommand(accRoleName, policyARN)
+
+		commands = append(commands, createRole, createPolicy, attachRolePolicy)
+	}
+
+	return awscb.JoinCommands(commands), nil
+}
+
+func (up *unmanagedPoliciesCreator) getRoleTags(roleType string, input *accountRolesCreationInput) map[string]string {
+	return getBaseRoleTags(roleType, input)
+}
+
+type hcpUnmanagedPoliciesCreator struct{}
+
+func (hcp *hcpUnmanagedPoliciesCreator) createRoles(r *rosa.Runtime, input *accountRolesCreationInput) error {
+	for file, role := range aws.HCPAccountRoles {
+		accRoleName := aws.GetRoleName(input.prefix, role.Name)
+		if !confirm.Prompt(true, "Create the '%s' role?", accRoleName) {
+			continue
 		}
 
-		r.Reporter.Debugf("Attaching permission policy to role '%s'", filename)
-		err = r.AWSClient.AttachRolePolicy(accRoleName, policyARN)
+		assumeRolePolicy := getAssumeRolePolicy(file, input)
+		tagsList := hcp.getRoleTags(file, input)
+		filename := fmt.Sprintf("sts_hcp_%s_permission_policy", file)
```

Comment:
The `_hcp` infix differentiates the classic rosa permission policies from the Hypershift permission policies.
SRE is updating the file names here https://github.com/openshift/managed-cluster-config/pull/1525 to match this convention.

Reply:
This was what I was going to ask about, so you are set already but there's still work to do somewhere else. Did you download everything and renamed to test then?

Reply:
Yes, I created the files locally to test the changes.
Changed the PR to draft until changes in managed cluster config are done, thanks.